lib/dao.js

```diff
@@ -2689,7 +2689,23 @@ DataAccessObject.updateAll = function(where, data, options, cb) {
     Model.notifyObserversOf('before save', context,
       function(err, ctx) {
         if (err) return cb(err);
-        doUpdate(ctx.where, ctx.data);
+
+        data = ctx.data;
+        var update = data;
```

Comment:
The whole point is to keep the `data` variable clean. If you scroll down along the lines, after making a copy of `data` into `update`, we modify `update` so that the data returned from the **before save** hook is not changed.
```js
        data = ctx.data;
        var update = data;
        var inst = data;
        if (!(data instanceof Model)) {
          inst = new Model(data, {applyDefaultValues: false});
        }
        update = inst.toObject(false);

        // validation required
        inst.isValid(function(valid) {
          if (valid) {
            doUpdate(ctx.where, ctx.data);
          } else {
            cb(new ValidationError(inst), inst);
          }
        }, update, options);
      });
```